src/main/java/com/arpnetworking/tsdcore/sinks/DimensionInjectingSink.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.sinks;
+
+import com.arpnetworking.tsdcore.model.DefaultKey;
+import com.arpnetworking.tsdcore.model.PeriodicData;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import net.sf.oval.constraint.NotNull;
+
+import java.util.Map;
+
+/**
+ * Sink adds any specified dimensions.
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
+ */
+public final class DimensionInjectingSink extends BaseSink {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void recordAggregateData(final PeriodicData data) {
+        final Map<String, String> mergedDimensions = Maps.newHashMap(data.getDimensions().getParameters());
+        mergedDimensions.putAll(_dimensions);
```

Comment:
Should this force override the dimensions in the data or only inject values for dimensions that don't already exist?  I could see it either way but would probably default to not overriding.

Reply:
I agree.  We should make it configurable on whether or not to overwrite and should probably not overwrite by default.

Reply:
Done and done. I've separated the configuration into `defaultDimensions` and `overrideDimensions` and inject this into the union map between the data dimensions. Specifically, data overrides defaults and overrides overwrite both.